roms_tools/setup/cdr_forcing.py

```diff
@@ -503,136 +285,227 @@ def plot_volume_flux(self, start=None, end=None, releases="all"):
             Start datetime for the plot. If None, defaults to `self.start_time`.
         end : datetime or None
             End datetime for the plot. If None, defaults to `self.end_time`.
-        releases : str, list of str, or "all", optional
-            A string or list of release names to plot.
+        release_names : list of str, or "all", optional
+            A list of release names to plot.
             If "all", the method will plot all releases.
             The default is "all".
+
+        Raises
+        ------
+        ValueError
+            If self.releases are not of type VolumeRelease.
+            If `release_names` is not a list of strings or "all".
+            If any of the specified release names do not exist in `self.releases`.
         """
 
+        if self.release_type != VolumeRelease:
+            raise ValueError(
+                "plot_volume_flux is only supported when all releases are of type VolumeRelease."
+            )
+
         start = start or self.start_time
         end = end or self.end_time
 
-        # Handle "all" releases case
-        if releases == "all":
-            releases = [k for k in self.releases if k != "_tracer_metadata"]
-        # Validate input for release names
-        self._validate_release_input(releases)
+        valid_release_names = [r.name for r in self.releases]
+
+        if release_names == "all":
+            release_names = valid_release_names
+
+        _validate_release_input(release_names, valid_release_names)
 
         data = self.ds["cdr_volume"]
 
         self._plot_line(
             data,
-            releases,
+            release_names,
             start,
             end,
             title="Volume flux of release(s)",
             ylabel=r"m$^3$/s",
```

Comment:
Does ylabel need to be a parameter? I thought we added a "units" field to the dataset previously?

Reply:
The `ylabel` here has nicer LaTeX formatting.